core/indexer/dataDispatcher.go

```diff
@@ -0,0 +1,105 @@
+package indexer
+
+import (
+	"context"
+	"time"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core/indexer/workItems"
+)
+
+var log = logger.GetOrCreate("core/indexer")
+
+// Options structure holds the indexer's configuration options
+type Options struct {
+	IndexerCacheSize  int
+	UseKibana         bool
+	TxIndexingEnabled bool
+}
+
+const (
+	backOffTime = time.Second * 10
+	maxBackOff  = time.Minute * 5
+)
+
+type dataDispatcher struct {
+	backOffTime   time.Duration
+	chanWorkItems chan workItems.WorkItemHandler
+	cancelFunc    func()
+}
+
+// NewDataDispatcher creates a new dataDispatcher instance, capable of selecting the correct es that will
+//  handle saving different types
+func NewDataDispatcher(cacheSize int) (*dataDispatcher, error) {
+	dd := &dataDispatcher{
+		chanWorkItems: make(chan workItems.WorkItemHandler, cacheSize),
+	}
+
+	return dd, nil
+}
+
+// StartIndexData will start index data in database
+func (d *dataDispatcher) StartIndexData() {
+	var ctx context.Context
+	ctx, d.cancelFunc = context.WithCancel(context.Background())
+
+	go d.startWorker(ctx)
+}
+
+func (d *dataDispatcher) startWorker(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			log.Debug("dispatcher's go routine is stopping...")
+			return
+		case wi := <-d.chanWorkItems:
+			d.doWork(wi)
+		}
+	}
+}
+
+// Close will close the endless running go routine
+func (d *dataDispatcher) Close() error {
+	if d.cancelFunc != nil {
+		d.cancelFunc()
+	}
+
+	return nil
+}
+
+// Add will add a new item in queue
+func (d *dataDispatcher) Add(item workItems.WorkItemHandler) {
+	d.chanWorkItems <- item
```

Comment:
Since this is an exported method you need to use `check.IfNil(item)` and make workItems.WorkItemHandler have the IsInterfaceNil method

Reply:
done.